test_runner/regress/test_ondemand_download.py

```diff
@@ -212,6 +213,9 @@ def get_resident_physical_size():
     log.info(filled_size)
     assert filled_current_physical == filled_size, "we don't yet do layer eviction"
 
+    # Wait until generated image layers are uploaded to S3
+    time.sleep(3)
```

Comment:
There's now a `wait_for_upload` helper as well instead of sleeps.

Reply:
`wait_for_upload` waits until specified LSN reaches S3, doesn't it?
But how it can enforce uploading of  generated by compaction image layer?
Right now this test performs:
```
    # wait until pageserver has successfully uploaded all the data to remote storage
    wait_for_sk_commit_lsn_to_reach_remote_storage(
        tenant_id, timeline_id, env.safekeepers, env.pageserver
    )
```

but it doesn't help.

Reply:
Oki I was just looking at the comment above the sleep and thought of `wait_for_upload`.